### DISCUSSION TITLE
RESTful protobuf handling

### DISCUSSION THREAD

**Discussion by raccettura** [home-assistant/architecture#620] [General] [answered]
HomeAssistant RESTful support is pretty awesome for consuming random arbitrary data from an assortment of things JSON or XML. However recently I've wanted to start getting some data available as protobuf.

It seems to be possible if I used `curl`, and piped through `protoc`, but that's cumbersome.  It would be nice if REST supported `application/protobuf` as a content-type and decoded it so that templates could easily digest the data further.

**Comment by balloob** [comment, marked as answer]
This is not common enough to add it to the rest sensor.
